examples/tutorial/x/blog/handlers_test.go

```diff
@@ -0,0 +1,466 @@
+				Text:   "We have created a room for live communication that is solely dedicated to high-level product discussions because this is a crucial support for fostering a technical user base within our broader community. Just as IOV is developing a full platform suite that includes retail products such as the universal wallet and B2B tools such as the BNS, each kind of community has a place in the movement toward mass adoption of blockchains which we aspire to lead.Another important reason that we established the #Developers room is that it provides a forum for users to receive help from our devs, and from each other, when playing with demos and live releases of IOV products in the future: as one can imagine, getting help with your test node or maintaining a highly dense conversation might be especially difficult in Telegram, depending on how many lambo memes and amusing gifs might be flying around at any given moment!We’re therefore happy to say that #Developers is launching with good timing — because community members who are interested in seeing our development progress for themselves can already try out our IOV-core release (read about it here!), and by the end of this month our public alphanet is launching! Keep your eyes open in coming weeks for this exciting release.",
+				Author: x.MainSigner(ctx, auth).Address(),
+			},
+			parent: CreateBlogMsg{
+				Slug:    "this_is_a_blog",
+				Title:   "this is a blog title",
+				Authors: [][]byte{x.MainSigner(ctx, auth).Address()},
+			},
+			res: weave.CheckResult{
+				GasAllocated: newPostCost,
+			},
+		},
+	}
+
+	for _, test := range testcases {
+		db := store.MemStore()
+		_, err := test.handler.Check(ctx, db, newTx(&test.msg))
+		require.EqualError(t, err, errBlogNotFound.Error())
+		newTestHandler("CreateBlogMsgHandler", auth).Deliver(ctx, db, newTx(&test.parent))
+		res, err := test.handler.Check(ctx, db, newTx(&test.msg))
+		require.NoError(t, err)
+		require.EqualValues(t, test.res, res, fmt.Sprintf("gas allocated cost was equal to %d", res.GasAllocated))
+	}
+}
+
+func TestCreatePostMsgHandlerDeliver(t *testing.T) {
+	newTx := x.TestHelpers{}.MockTx
+	ctx, auth := newContextWithAuth("3AFCDAB4CFBF066E959D139251C8F0EE91E99D5A")
+	testcases := []struct {
+		handler CreatePostMsgHandler
+		msg     CreatePostMsg
+		parent  CreateBlogMsg
+		obj     Post
+	}{
+		{
+			handler: newTestHandler("CreatePostMsgHandler", auth).(CreatePostMsgHandler),
+			msg: CreatePostMsg{
+				Blog:   "this_is_a_blog",
+				Title:  "this is a title",
+				Text:   "We have created a room for live communication that is solely dedicated to high-level product discussions because this is a crucial support for fostering a technical user base within our broader community. Just as IOV is developing a full platform suite that includes retail products such as the universal wallet and B2B tools such as the BNS, each kind of community has a place in the movement toward mass adoption of blockchains which we aspire to lead.",
```

Comment:
Same as above, pull long text into package-level variable.
Also, create the address at top of test function, rather than recreate it every time: `x.MainSigner(ctx, auth).Address()`